snapper/Logger.cc

```diff
@@ -37,10 +38,25 @@ namespace snapper
     using namespace std;
 
 
-    // Use a pointer to avoid a global destructor. Otherwise the Snapper
-    // destructor in Factory.cc can be called after when logging does not work
-    // anymore. TODO: nicer solution.
-    string* filename = new string("/var/log/snapper.log");
+    namespace
+    {
+
+	struct LoggerData
+	{
+	    LoggerData() : filename("/var/log/snapper.log"), mutex() {}
```

Comment:
NP: This string is hardcoded in several places, it would be nice to have a constant for it to have a single place to change. BTW the log file is not configurable?

Reply:
Added a define.

No, the log filename is not configurable. You would also need to modify logrotate in that case so it is not easy.